# Nashville, TN *the* Music City? 


### Introduction/Business Problem

Question: What city can really claim to be the Music City? 

Audience: Tourist Publications, Social Media, General Reputation 

Nashville, Tennesse has the claim of fame of being Music City USA. In the past few years Nashville has seen rapid expansion in many other industries. Many people compare Nashville, TN to Austiin, TX. Both cities are upcoming large southern metropolis, who also both claim their large music scenes. I want to use the data science methodology to try to prove its claim to fame on the world stage. If Nashville is able to say it is the definitive Music City of the United States, it will increase its reputation on the map even further bringing more tourist, more business, and more people to the city I call home. 

### Data Sources

I will leverage the Foursquare location data to find the number of music venues in each zipcode of the city, and then compare the amount of music venues, diversity of music played there, and overall ranking of the venues to the world's other city well known city for music, a classic Nashville rival: Austin, Texas. 

Foursquare Location Date:
Venue Type- Live Music- Will give me the respective totals of places to go see live music for each. 
Venue Rating- I will take the average rating of each location and compare the two cities music venues average rating. It gives further depth into the quality of the music venues. 
Venue Location- Will allow me to group the neighborhoods to where the music venues locations are. The distance between each music venue could give insights into the distribution of the designed environment.
Music Genre at the Music Venue? 

Nashville, TN:
Zip Code Information
Neighborhood Information

Austin, TX;
Zip Code Information
Neighborhood Information 


### Methodology

I decided to import easy CSV data from a few websites using Beautiful Soup to create the csv's. That is what gave me the zip code and neighborhood information for Austin and Nashville. 

Geocoder allowed me to get more percise and put the latitude and longitutde of each zipcode for Austin and Nashville. I then used Folium to make sure that I had put the zip-codes in the right latitude and longitude.

I then used the Foursquare API to get a specific data frame of all of the venues around the latitude and longitude of Austin and Nashville for places that were categorized as "Music Venues." I decided to keep that search very clear, so that I would not have to sort through a lot more data, and to only include places Foursquare would consider an actual music venue. I cleaned up the API results for both cities and merged those results to the zip code and neighborhood dataframe. 

Once I had both cities mapped out with where the music venues were and what neighborhoods they were in I used K nearest neighbor to identify any classifcations of data sets within my data. For Austin and Nashville I grouped by each Neighborhood and then Zip code- once I had them grouped I put the one-hot variables next to each venue to assign a clustering value.





### Results

The Foursquare API search for music venues returned 24 music venues for Nashville, and 13 for Austin. 

The KNN showed the distribution of the music venues throughout the neighborhoods of each city was relatively the same, both cities only had one neighborhood that stood out from the rest for its number of music venues.



### Conclusion

In [1]:
import requests 
import pandas as pd 
import numpy as np 
import random 
from IPython.display import Image 
from IPython.core.display import HTML 
from pandas.io.json import json_normalize
from geopy.geocoders import Nominatim
import matplotlib.cm as cm
import matplotlib.colors as colors
from sklearn.cluster import KMeans
!pip install folium
import folium
!pip install geocoder
import geocoder



     |████████████████████████████████| 92kB 10.2MB/s eta 0:00:01
     |████████████████████████████████| 102kB 5.9MB/s eta 0:00:01


### Nashville and Austin Neighborhood and Zipcode Data Set Up

In [2]:
# The code was removed by Watson Studio for sharing.

Neighborhood  Zip Code
0  Antioch/Cane Ridge     37013
1           Brentwood     37027
2      Goodlettsville     37072
3           Hermitage     37076
4             Madison     37115

In [3]:
lat = []
lng = []
lat_lng_coords = None

zip_code = Nashville['Zip Code']

for pc in zip_code:
    g = geocoder.arcgis('{}, Nashville, Tennessee'.format(pc))
    lat_lng_coords = g.latlng
    lat.append(lat_lng_coords[0])
    lng.append(lat_lng_coords[1])
   
Nashville2 = Nashville
Nashville2['Latitude'] = lat
Nashville2['Longitude'] = lng

Nashville2

Neighborhood  Zip Code   Latitude  Longitude
0   Antioch/Cane Ridge     37013  36.047051 -86.650867
1            Brentwood     37027  36.052222 -86.769434
2       Goodlettsville     37072  36.302708 -86.695278
3            Hermitage     37076  36.180321 -86.600227
4              Madison     37115  36.241080 -86.737137
5          Old Hickory     37138  36.213912 -86.585522
6                SoBro     37201  36.165885 -86.777900
7             West End     37203  36.148965 -86.780381
8             12 South     37204  36.110095 -86.776305
9          Belle Meade     37205  36.108823 -86.877707
10      East Nashville     37206  36.182265 -86.731055
11     North Nashville     37207  36.228175 -86.781770
12          Germantown     37208  36.176155 -86.803740
13         Sylvan Park     37209  36.146195 -86.857702
14            Woodbine     37210  36.137455 -86.732810
15       Souh Nashille     37211  36.061331 -86.720030
16           Music Row     37212  36.132818 -86.803792
17            Donelson     37214  36.169560 -86.671400
18         Green Hills     37215  36.079859 -86.840510
19           Inglewood     37216  36.216063 -86.726874
20      Briley Parkway     37217  36.107370 -86.691526
21            Bordeaux     37218  36.203940 -86.851497
22            Downtown     37219  36.170400 -86.784060
23            Oak Hill     37220  36.070320 -86.783555
24            Bellevue     37221  36.059790 -86.964720

In [4]:
g = geocoder.arcgis('Nashville, Tennessee')
lat_Nash = g.latlng[0]
lng_Nash = g.latlng[1]
Nashvillemap = folium.Map(location=[lat_Nash, lng_Nash], zoom_start=11)
for lat, lng, bor, zip_code in zip(Nashville2['Latitude'], Nashville2['Longitude'], Nashville2['Neighborhood'], Nashville2['Zip Code']):
    label = '{}'.format(zip_code)       
    label = folium.Popup(label, parse_html=True)
    folium.CircleMarker([lat, lng],
                        radius=5,
                        popup=label,
                        color='blue',
                        fill=True,
                        fill_color='#3186cc',
                        fill_opacity=0.7,
                        parse_html=False).add_to(Nashvillemap)
    
Nashvillemap

In [81]:

body = client_f4061d98cb894b639d1da1394688a9a2.get_object(Bucket='craftcourseracapstone-donotdelete-pr-jhwzebigv05uwq',Key='Austin Texas.xlsx')['Body']

if not hasattr(body, "__iter__"): body.__iter__ = types.MethodType( __iter__, body )

Austin = pd.read_excel(body)
Austin


Neighborhood  Zipcode
0        East Side    78702
1        Tarrytown    78703
2   South Congress    78704
3          Mueller    78723
4        Hyde Park    78731
5      Lake Travis    78732
6         Bee Cave    78738
7         Westlake    78746
8    Sunset Valley    78745
9   Southland Oaks    78748
10        Oak Park    78749
11   Anderson Mill    78750
12       Allandale    78757
13     Great Hills    78759

In [87]:
g = geocoder.arcgis('Austin, Texas')
lat = []
lng = []
lat_lng_coords = None

zip_code = Austin['Zipcode']

for pc in zip_code:
    g = geocoder.arcgis('{}, Austin, Texas'.format(pc))
    lat_lng_coords = g.latlng
    lat.append(lat_lng_coords[0])
    lng.append(lat_lng_coords[1])
   
Austin2 = Austin
Austin2['Latitude'] = lat
Austin2['Longitude'] = lng
Austin2

Neighborhood  Zipcode   Latitude  Longitude
0        East Side    78702  30.264515 -97.715185
1        Tarrytown    78703  30.291375 -97.766235
2   South Congress    78704  30.241225 -97.769465
3          Mueller    78723  30.304778 -97.684570
4        Hyde Park    78731  30.345875 -97.771394
5      Lake Travis    78732  30.381990 -97.894040
6         Bee Cave    78738  30.307911 -97.991514
7         Westlake    78746  30.293652 -97.821940
8    Sunset Valley    78745  30.200750 -97.800080
9   Southland Oaks    78748  30.151714 -97.825746
10        Oak Park    78749  30.205503 -97.857410
11   Anderson Mill    78750  30.416503 -97.805708
12       Allandale    78757  30.352935 -97.734685
13     Great Hills    78759  30.395913 -97.750012

In [83]:
g = geocoder.arcgis('Austin, Texas')
lat_Austin = g.latlng[0]
lng_Austin = g.latlng[1]
Austinmap = folium.Map(location=[lat_Austin, lng_Austin], zoom_start=11)
for lat, lng, bor, zip_code in zip(Austin2['Latitude'], Austin2['Longitude'], Austin2['Neighborhood'], Austin2['Zipcode']):
    label = '{}'.format(zip_code)       
    label = folium.Popup(label, parse_html=True)
    folium.CircleMarker([lat, lng],
                        radius=5,
                        popup=label,
                        color='blue',
                        fill=True,
                        fill_color='#3186cc',
                        fill_opacity=0.7,
                        parse_html=False).add_to(Nashvillemap)
    
Austinmap

### API Setup- Nashville

In [31]:
CLIENT_ID='FQROVQZIVWZEWWMHFEVMNFBMOTPUDRBD2RTTLKZAYVGW4PC4'
CLIENT_SECRET='LS2UHM2OU2BX3JDY4ZR04RMX4QJ0EH2MEUSLDQ0IXH4YSRMZ'
VERSION= '20191007'
LIMIT= 500
categoryId= '4bf58dd8d48988d1e5931735'
Latitude= 36.1627
Longitude= (-86.7816)


In [68]:
url = 'https://api.foursquare.com/v2/venues/search?client_id={}&client_secret={}&ll={},{}&v={}&categoryId={}&radius={}&limit={}'.format(CLIENT_ID, CLIENT_SECRET, Latitude, Longitude, VERSION, categoryId, radius, LIMIT)
url

'https://api.foursquare.com/v2/venues/search?client_id=FQROVQZIVWZEWWMHFEVMNFBMOTPUDRBD2RTTLKZAYVGW4PC4&client_secret=LS2UHM2OU2BX3JDY4ZR04RMX4QJ0EH2MEUSLDQ0IXH4YSRMZ&ll=30.2672,-97.7431&v=20191007&categoryId=4bf58dd8d48988d1e5931735&radius=10000&limit=500'

In [70]:
results = requests.get(url).json()


# assign relevant part of JSON to venues
venues = results['response']['venues']

# tranform venues into a dataframe
NashvilleVenues = json_normalize(venues)
NashvilleVenues.head()



categories delivery.id  \
0  [{'id': '4bf58dd8d48988d116941735', 'name': 'B...         NaN   
1  [{'id': '4bf58dd8d48988d1e5931735', 'name': 'M...         NaN   
2  [{'id': '4bf58dd8d48988d1e5931735', 'name': 'M...         NaN   
3  [{'id': '4bf58dd8d48988d1e7931735', 'name': 'J...         NaN   
4  [{'id': '4bf58dd8d48988d1df931735', 'name': 'B...         NaN   

  delivery.provider.icon.name delivery.provider.icon.prefix  \
0                         NaN                           NaN   
1                         NaN                           NaN   
2                         NaN                           NaN   
3                         NaN                           NaN   
4                         NaN                           NaN   

  delivery.provider.icon.sizes delivery.provider.name delivery.url  hasPerk  \
0                          NaN                    NaN          NaN    False   
1                          NaN                    NaN          NaN    False   
2                          NaN                    NaN          NaN    False   
3                          NaN                    NaN          NaN    False   
4                          NaN                    NaN          NaN    False   

                         id        location.address  ...  \
0  49b6d215f964a52013531fe3  422 Congress Ave Ste D  ...   
1  40b13b00f964a520e8f31ee3        412 Congress Ave  ...   
2  4ba452f8f964a520fe9538e3           604 Brazos St  ...   
3  4101a680f964a520640b1fe3        315 Congress Ave  ...   
4  5676037b498e1b7d98c935c7        217 Congress Ave  ...   

                           location.formattedAddress  \
0  [422 Congress Ave Ste D (at 5th St), Austin, T...   
1  [412 Congress Ave (btwn 4th & 5th St.), Austin...   
2   [604 Brazos St, Austin, TX 78701, United States]   
3  [315 Congress Ave (at 3rd St.), Austin, TX 787...   
4  [217 Congress Ave (at 2nd St), Austin, TX 7870...   

                             location.labeledLatLngs location.lat  \
0  [{'label': 'display', 'lat': 30.2671262965541,...    30.267126   
1  [{'label': 'display', 'lat': 30.26666907980697...    30.266669   
2  [{'label': 'display', 'lat': 30.26774749787490...    30.267747   
3  [{'label': 'display', 'lat': 30.26567161471170...    30.265672   
4  [{'label': 'display', 'lat': 30.26496553144536...    30.264966   

  location.lng  location.neighborhood location.postalCode location.state  \
0   -97.743322                    NaN               78701             TX   
1   -97.743427                    NaN               78701             TX   
2   -97.741654                    NaN               78701             TX   
3   -97.743575                    NaN               78701             TX   
4   -97.743747                    NaN               78701             TX   

                              name    referralId venuePage.id  
0                  Shiner's Saloon  v-1570560772     63571371  
1                        Speakeasy  v-1570560772     45192438  
2          Driskill Victorian Room  v-1570560772          NaN  
3                The Elephant Room  v-1570560772          NaN  
4  Cooper's Old Time Pit Bar-B-Que  v-1570560772          NaN  

[5 rows x 25 columns]

In [71]:
NashvilleVenues2 = ['name', 'categories'] + [col for col in dataframe.columns if col.startswith('location.')] + ['id']
NashvilleVenues2 = dataframe.loc[:, NashvilleVenues2]


def get_category_type(row):
    try:
        categories_list = row['categories']
    except:
        categories_list = row['venue.categories']
        
    if len(categories_list) == 0:
        return None
    else:
        return categories_list[0]['name']


NashvilleVenues2['categories'] = NashvilleVenues2.apply(get_category_type, axis=1)


NashvilleVenues2.columns = [column.split('.')[-1] for column in NashvilleVenues2.columns]

NashvilleVenues2

name              categories  \
0            That's CooL Nashville A Live Music Venue               Speakeasy   
1                    Live On The Green Music Festival             Music Venue   
2                                   Music City Center       Convention Center   
3               Country Music Hall of Fame and Museum                  Museum   
4                                   Renaissance Hotel                   Hotel   
5                        Music City Walk of Fame Park                    Park   
6         The Grand Ballroom At The Music City Center       Convention Center   
7                           Sony/ATV music publishing                  Office   
8                Music City Star - Riverfront Station           Train Station   
9                                 Music City Showcase           Souvenir Shop   
10                    Renaissance Music City Ballroom                   Hotel   
11                                 Music City Central             Bus Station   
12                   country music hall of fame store           Souvenir Shop   
13                                  City of Nashville                    City   
14                       Republic Live Administration                  Office   
15                         The Pink Pistol Music City                Boutique   
16                                   Ryman Auditorium             Music Venue   
17                                    2nd Avenue Live                     Bar   
18                                     Nashville Live               Rock Club   
19                                  Music City Events                Building   
20                 Music City Trolley HOP on/off Tour  Transportation Service   
21                         New Faces of Country Music            Concert Hall   
22                  Just Love Coffee Cafe - Music Row          Breakfast Spot   
23                   Music City Center Parking Garage                 Parking   
24                                Music City Networks                  Office   
25   Siriusxm Music City Theatre at Bridgestone Arena           Indie Theater   
26                                  Music City Crepes                Creperie   
27                        Best Western Plus Music Row                   Hotel   
28  Songwriter Session at the Country Music Hall o...                    None   
29                       Sam A. Wilson Music Building   College Arts Building   
30                                  Element Music Row      Real Estate Office   
31                                Music City Fine Art             Art Gallery   
32                        Music City Rollin' Jamboree             Music Venue   
33          National Museum of African American Music                  Museum   
34                              Universal Music Group         Conference Room   
35                                    Music City Bowl        Football Stadium   
36                                     Carnival Music                  Office   
37                  Music City Central Parking Garage                 Parking   
38                                           Live Oak                     Bar   
39                              Music City Chicken Co     Fried Chicken Joint   
40           Music City Circuit - Blue Route Downtown                Bus Line   
41                                          Music Row             Music Venue   
42                                Music City Ballroom            Meeting Room   
43                                     The Music Loft             Music Venue   
44                   Music City Circuit - Green Route                Bus Line   
45                                     Music City Cab                    Taxi   
46                      Grimey's New & Preloved Music             Record Shop   
47                         Tailgate Brewery Music Row                 Brewery   
48                    Exhibit Hal A Music City Center       Convention Center   

In [72]:
frames= [NashvilleVenues2, Nashville2]

NashvilleMerged = pd.concat(frames, axis= 1, join= 'inner', sort=False)

In [96]:
NashvilleMerged.drop(['lat','lng','neighborhood','postalCode','crossStreet'], axis=1, inplace= True)
NashvilleMerged

name              categories  \
0      That's CooL Nashville A Live Music Venue               Speakeasy   
1              Live On The Green Music Festival             Music Venue   
2                             Music City Center       Convention Center   
3         Country Music Hall of Fame and Museum                  Museum   
4                             Renaissance Hotel                   Hotel   
5                  Music City Walk of Fame Park                    Park   
6   The Grand Ballroom At The Music City Center       Convention Center   
7                     Sony/ATV music publishing                  Office   
8          Music City Star - Riverfront Station           Train Station   
9                           Music City Showcase           Souvenir Shop   
10              Renaissance Music City Ballroom                   Hotel   
11                           Music City Central             Bus Station   
12             country music hall of fame store           Souvenir Shop   
13                            City of Nashville                    City   
14                 Republic Live Administration                  Office   
15                   The Pink Pistol Music City                Boutique   
16                             Ryman Auditorium             Music Venue   
17                              2nd Avenue Live                     Bar   
18                               Nashville Live               Rock Club   
19                            Music City Events                Building   
20           Music City Trolley HOP on/off Tour  Transportation Service   
21                   New Faces of Country Music            Concert Hall   
22            Just Love Coffee Cafe - Music Row          Breakfast Spot   
23             Music City Center Parking Garage                 Parking   
24                          Music City Networks                  Office   

                              address  cc       city        country  distance  \
0                 2309B Franklin Pike  US  Nashville  United States      3930   
1                         1 Public Sq  US  Nashville  United States       516   
2                       201 5th Ave S  US  Nashville  United States       665   
3                       222 5th Ave S  US  Nashville  United States       670   
4                     611 Commerce St  US  Nashville  United States       185   
5                       121 4th Ave S  US  Nashville  United States       586   
6                                 NaN  US  Nashville  United States       698   
7                       424 Church St  US  Nashville  United States       169   
8                       108 1st Ave S  US  Nashville  United States       734   
9                        323 Broadway  US  Nashville  United States       451   
10                                NaN  US  Nashville  United States       183   
11                  400 Charlotte Ave  US  Nashville  United States       449   
12                                NaN  US  Nashville  United States       641   
13                                NaN  US        NaN  United States       330   
14                                NaN  US  Nashville  United States        99   
15                      218 3rd Ave N  US  Nashville  United States       388   
16                      116 5th Ave N  US  Nashville  United States       307   
17                      154 2nd Ave N  US  Nashville  United States       528   
18                      131 2nd Ave N  US  Nashville  United States       501   
19  808 Broadway, Nashville, TN 37203  US  Nashville  United States       549   
20                         5th Avenue  US  Nashville  United States       609   
21                    Commerce street  US  Nashville  United States       116   
22                 1528 Demonbreun St  US  Nashville  United States      1364   
23                                NaN  US  Nashville  United States       817   
24                     209 10th Ave S  US  Nashville  United States

### API Setup- Austin

In [64]:
CLIENT_ID='FQROVQZIVWZEWWMHFEVMNFBMOTPUDRBD2RTTLKZAYVGW4PC4'
CLIENT_SECRET='LS2UHM2OU2BX3JDY4ZR04RMX4QJ0EH2MEUSLDQ0IXH4YSRMZ'
VERSION= '20191007'
LIMIT= 500
categoryId= '4bf58dd8d48988d1e5931735'
Latitude2= 30.2672
Longitude2= (-97.7431)

In [75]:
url2 = 'https://api.foursquare.com/v2/venues/search?client_id={}&client_secret={}&ll={},{}&v={}&categoryId={}&radius={}&limit={}'.format(CLIENT_ID, CLIENT_SECRET, Latitude2, Longitude2, VERSION, categoryId, radius, LIMIT)
results2 = requests.get(url2).json()

venues2 = results2['response']['venues']

AustinVenues = json_normalize(venues2)

AustinVenues

categories delivery.id  \
0   [{'id': '4bf58dd8d48988d116941735', 'name': 'B...         NaN   
1   [{'id': '4bf58dd8d48988d1e5931735', 'name': 'M...         NaN   
2   [{'id': '4bf58dd8d48988d1e5931735', 'name': 'M...         NaN   
3   [{'id': '4bf58dd8d48988d1e7931735', 'name': 'J...         NaN   
4   [{'id': '4bf58dd8d48988d1df931735', 'name': 'B...         NaN   
5   [{'id': '4bf58dd8d48988d121941735', 'name': 'L...         NaN   
6   [{'id': '4bf58dd8d48988d17f941735', 'name': 'M...         NaN   
7   [{'id': '4bf58dd8d48988d1e5931735', 'name': 'M...         NaN   
8   [{'id': '4bf58dd8d48988d116941735', 'name': 'B...         NaN   
9   [{'id': '4bf58dd8d48988d1e5931735', 'name': 'M...         NaN   
10  [{'id': '4bf58dd8d48988d1e5931735', 'name': 'M...         NaN   
11  [{'id': '4bf58dd8d48988d117941735', 'name': 'B...         NaN   
12  [{'id': '4bf58dd8d48988d121941735', 'name': 'L...         NaN   
13  [{'id': '4bf58dd8d48988d1df931735', 'name': 'B...         NaN   
14  [{'id': '4bf58dd8d48988d14e941735', 'name': 'A...         NaN   
15  [{'id': '4bf58dd8d48988d116941735', 'name': 'B...         NaN   
16  [{'id': '4bf58dd8d48988d1e5931735', 'name': 'M...         NaN   
17  [{'id': '4bf58dd8d48988d1e9931735', 'name': 'R...         NaN   
18  [{'id': '4bf58dd8d48988d1e5931735', 'name': 'M...         NaN   
19  [{'id': '4bf58dd8d48988d1e5931735', 'name': 'M...         NaN   
20  [{'id': '4bf58dd8d48988d1e5931735', 'name': 'M...         NaN   
21  [{'id': '4bf58dd8d48988d1e9931735', 'name': 'R...         NaN   
22  [{'id': '4bf58dd8d48988d116941735', 'name': 'B...         NaN   
23  [{'id': '4bf58dd8d48988d1e5931735', 'name': 'M...         NaN   
24  [{'id': '4bf58dd8d48988d1e9931735', 'name': 'R...         NaN   
25  [{'id': '4bf58dd8d48988d11f941735', 'name': 'N...         NaN   
26  [{'id': '4bf58dd8d48988d1e9931735', 'name': 'R...         NaN   
27  [{'id': '4bf58dd8d48988d1e9931735', 'name': 'R...         NaN   
28  [{'id': '4bf58dd8d48988d1e8931735', 'name': 'P...         NaN   
29  [{'id': '4bf58dd8d48988d116941735', 'name': 'B...         NaN   
30  [{'id': '4bf58dd8d48988d1e5931735', 'name': 'M...         NaN   
31  [{'id': '4bf58dd8d48988d118941735', 'name': 'D...         NaN   
32  [{'id': '4bf58dd8d48988d129941735', 'name': 'C...         NaN   
33  [{'id': '4bf58dd8d48988d1e9931735', 'name': 'R...      650927   
34  [{'id': '4bf58dd8d48988d118941735', 'name': 'D...         NaN   
35  [{'id': '4bf58dd8d48988d116941735', 'name': 'B...         NaN   
36  [{'id': '4bf58dd8d48988d132941735', 'name': 'C...         NaN   
37  [{'id': '4bf58dd8d48988d1e9931735', 'name': 'R...         NaN   
38  [{'id': '4bf58dd8d48988d1e9931735', 'name': 'R...         NaN   
39  [{'id': '4bf58dd8d48988d1e5931735', 'name': 'M...         NaN   
40  [{'id': '4bf58dd8d48988d1e5931735', 'name': 'M...         NaN   
41  [{'id': '4bf58dd8d48988d1e5931735', 'name': 'M...         NaN   
42  [{'id': '4bf58dd8d48988d1e5931735', 'name': 'M...         NaN   
43  [{'id': '4bf58dd8d48988d1e5931735', 'name': 'M...         NaN   
44  [{'id': '4bf58dd8d48988d116941735', 'name': 'B...         NaN   
45  [{'id': '4bf58dd8d48988d121941735', 'name': 'L...         NaN   
46  [{'id': '4bf58dd8d48988d1e0931735', 'name': 'C...      456949   
47  [{'id': '4bf58dd8d48988d1e5931735', 'name': 'M...         NaN   
48  [{'id': '4bf58dd8d48988d116941735', 'name': 'B...         NaN   
49  [{'id': '4bf58dd8d48988d137941735', 'name': 'T...         NaN   

                delivery.provider.icon.name  \
0                                       NaN   
1                                       NaN   
2                                       NaN   
3                                       NaN   
4                                       NaN   
5                                       NaN   
6                                       NaN   
7                                       NaN   
8                                       NaN   
9                                       NaN   
10    

In [78]:
AustinVenues2 = ['name', 'categories'] + [col for col in dataframe.columns if col.startswith('location.')] + ['id']
AustinVenues2 = AustinVenues.loc[:, AustinVenues2]


def get_category_type(row):
    try:
        categories_list = row['categories']
    except:
        categories_list = row['venue.categories']
        
    if len(categories_list) == 0:
        return None
    else:
        return categories_list[0]['name']


AustinVenues2['categories'] = AustinVenues2.apply(get_category_type, axis=1)


AustinVenues2.columns = [column.split('.')[-1] for column in AustinVenues2.columns]

AustinVenues2

name           categories  \
0                                     Shiner's Saloon                  Bar   
1                                           Speakeasy          Music Venue   
2                             Driskill Victorian Room          Music Venue   
3                                   The Elephant Room            Jazz Club   
4                     Cooper's Old Time Pit Bar-B-Que            BBQ Joint   
5                              Cedar Street Courtyard               Lounge   
6                                   Paramount Theatre        Movie Theater   
7                                  Barracuda Backyard          Music Venue   
8                                 The Chuggin' Monkey                  Bar   
9                               Come And Take It Live          Music Venue   
10                          Santa Cruz Studio Theater          Music Venue   
11                                           The ABGB          Beer Garden   
12                                       The Townsend               Lounge   
13                                    Stubb's Bar-B-Q            BBQ Joint   
14                                        Geraldine's  American Restaurant   
15                                      Dirty Dog Bar                  Bar   
16                                 Anderson Band Hall          Music Venue   
17                                         Emo's East            Rock Club   
18                                       Darwin's Pub          Music Venue   
19                              Austin Soundwaves MLK          Music Venue   
20                                        Cactus Cafe          Music Venue   
21                                         The Parish            Rock Club   
22                                          Scoot Inn                  Bar   
23  Earl Campbell Bar, Austin-Bergstrom Internatio...          Music Venue   
24                               The Continental Club            Rock Club   
25                                           Antone's            Nightclub   
26                                  604 E. 6th Street            Rock Club   
27                                         The Mohawk            Rock Club   
28                           Pete's Dueling Piano Bar            Piano Bar   
29                                 Cheer Up Charlie's                  Bar   
30                                 Guero's Oak Garden          Music Venue   
31                     Ginny's Little Longhorn Saloon             Dive Bar   
32                                   Austin City Hall            City Hall   
33                                           Valhalla            Rock Club   
34                                       Donn's Depot             Dive Bar   
35                                          Swan Dive                  Bar   
36                        Central Presbyterian Church               Church   
37                                   Hole In The Wall            Rock Club   
38                                          Saxon Pub            Rock Club   
39                       3Ten Austin City Limits Live          Music Venue   
40                                          Barracuda          Music Venue   
41                                Empire Control Room          Music Venue   
42                  Stubb's Waller Creek Amphitheatre          Music Venue   
43                                  Music Lab St Elmo          Music Venue   
44                                     Skylark Lounge                  Bar   
45                                     The North Door               Lounge   
46                                   Kick Butt Coffee          Coffee Shop   
47                                        Hotel Vegas          Music Venue   
48                                  The Blind Pig Pub                  Bar   
49                              Scottish Rite Theatre              Theater   

                              address  cc    city        country  \
0 

In [89]:
frames2= [AustinVenues2, Austin2]

AustinMerged = pd.concat(frames2, axis= 1, join= 'inner', sort=False)

AustinMerged

name     categories                   address  \
0                   Shiner's Saloon            Bar    422 Congress Ave Ste D   
1                         Speakeasy    Music Venue          412 Congress Ave   
2           Driskill Victorian Room    Music Venue             604 Brazos St   
3                 The Elephant Room      Jazz Club          315 Congress Ave   
4   Cooper's Old Time Pit Bar-B-Que      BBQ Joint          217 Congress Ave   
5            Cedar Street Courtyard         Lounge              208 W 4th St   
6                 Paramount Theatre  Movie Theater          713 Congress Ave   
7                Barracuda Backyard    Music Venue              611 E 7th St   
8               The Chuggin' Monkey            Bar              219 E 6th St   
9             Come And Take It Live    Music Venue       2015 E Riverside Dr   
10        Santa Cruz Studio Theater    Music Venue             1805 E 7th St   
11                         The ABGB    Beer Garden          1305 W Oltorf St   
12                     The Townsend         Lounge  718 Congress Ave Ste 100   
13                  Stubb's Bar-B-Q      BBQ Joint          801 Red River St   

    cc    city        country                         crossStreet  distance  \
0   US  Austin  United States                           at 5th St        22   
1   US  Austin  United States                  btwn 4th & 5th St.        66   
2   US  Austin  United States                                 NaN       151   
3   US  Austin  United States                          at 3rd St.       176   
4   US  Austin  United States                           at 2nd St       256   
5   US  Austin  United States                   Colorado & Lavaca       216   
6   US  Austin  United States                             7th St.       269   
7   US  Austin  United States                                 NaN       646   
8   US  Austin  United States  btw Brazos St. & San Jacinto Blvd.       231   
9   US  Austin  United States                                 NaN      3225   
10  US  Austin  United States                                 NaN      2007   
11  US  Austin  United States             'Oltorf and the tracks'      3487   
12  US  Austin  United States                              8th St       275   
13  US  Austin  United States                         at E 8th St       674   

                                     formattedAddress  \
0   [422 Congress Ave Ste D (at 5th St), Austin, T...   
1   [412 Congress Ave (btwn 4th & 5th St.), Austin...   
2    [604 Brazos St, Austin, TX 78701, United States]   
3   [315 Congress Ave (at 3rd St.), Austin, TX 787...   
4   [217 Congress Ave (at 2nd St), Austin, TX 7870...   
5   [208 W 4th St (Colorado & Lavaca), Austin, TX ...   
6   [713 Congress Ave (7th St.), Austin, TX 78701,...   
7           [611 E 7th St, Austin, TX, United States]   
8   [219 E 6th St (btw Brazos St. & San Jacinto Bl...   
9   [2015 E Riverside Dr, Austin, TX 78741, United...   
10         [1805 E 7th St, Austin, TX, United States]   
11  [1305 W Oltorf St ('Oltorf and the tracks'), A...   
12  [718 Congress Ave Ste 100 (8th St), Austin, TX...   
13  [801 Red River St (at E 8th St), Austin, TX 78...   

                                       labeledLatLngs        lat        lng  \
0   [{'label': 'display', 'lat': 30.2671262965541,...  30.267126 -97.743322   
1   [{'label': 'display', 'lat': 30.26666907980697...  30.266669 -97.743427   
2   [{'label': 'display', 'lat': 30.26774749787490...  30.267747 -97.741654   
3   [{'label': 'display', 'lat': 30.26567161471170...  30.265672 -97.743575   
4   [{'label': 'display', 'lat': 30.26496553144536...  30.264966 -97.743747   
5   [{'label': 'display', 'lat': 30.26682931530129...  30.266829 -97.745313   
6   [{'label': 'display', 'lat': 30.26945691311456...  30.269457 -97.742077   
7   [{'label': 'display', 'lat': 30.26703478951284...  30.267035 -97.736379   
8   [{'label': 'display', 'lat': 30.26741007051066...  30.267410 -97.740701  

In [97]:
AustinMerged.drop(['lat','lng','neighborhood','postalCode','crossStreet'], axis=1, inplace= True)
AustinMerged

name     categories                   address  \
0                   Shiner's Saloon            Bar    422 Congress Ave Ste D   
1                         Speakeasy    Music Venue          412 Congress Ave   
2           Driskill Victorian Room    Music Venue             604 Brazos St   
3                 The Elephant Room      Jazz Club          315 Congress Ave   
4   Cooper's Old Time Pit Bar-B-Que      BBQ Joint          217 Congress Ave   
5            Cedar Street Courtyard         Lounge              208 W 4th St   
6                 Paramount Theatre  Movie Theater          713 Congress Ave   
7                Barracuda Backyard    Music Venue              611 E 7th St   
8               The Chuggin' Monkey            Bar              219 E 6th St   
9             Come And Take It Live    Music Venue       2015 E Riverside Dr   
10        Santa Cruz Studio Theater    Music Venue             1805 E 7th St   
11                         The ABGB    Beer Garden          1305 W Oltorf St   
12                     The Townsend         Lounge  718 Congress Ave Ste 100   
13                  Stubb's Bar-B-Q      BBQ Joint          801 Red River St   

    cc    city        country  distance  \
0   US  Austin  United States        22   
1   US  Austin  United States        66   
2   US  Austin  United States       151   
3   US  Austin  United States       176   
4   US  Austin  United States       256   
5   US  Austin  United States       216   
6   US  Austin  United States       269   
7   US  Austin  United States       646   
8   US  Austin  United States       231   
9   US  Austin  United States      3225   
10  US  Austin  United States      2007   
11  US  Austin  United States      3487   
12  US  Austin  United States       275   
13  US  Austin  United States       674   

                                     formattedAddress  \
0   [422 Congress Ave Ste D (at 5th St), Austin, T...   
1   [412 Congress Ave (btwn 4th & 5th St.), Austin...   
2    [604 Brazos St, Austin, TX 78701, United States]   
3   [315 Congress Ave (at 3rd St.), Austin, TX 787...   
4   [217 Congress Ave (at 2nd St), Austin, TX 7870...   
5   [208 W 4th St (Colorado & Lavaca), Austin, TX ...   
6   [713 Congress Ave (7th St.), Austin, TX 78701,...   
7           [611 E 7th St, Austin, TX, United States]   
8   [219 E 6th St (btw Brazos St. & San Jacinto Bl...   
9   [2015 E Riverside Dr, Austin, TX 78741, United...   
10         [1805 E 7th St, Austin, TX, United States]   
11  [1305 W Oltorf St ('Oltorf and the tracks'), A...   
12  [718 Congress Ave Ste 100 (8th St), Austin, TX...   
13  [801 Red River St (at E 8th St), Austin, TX 78...   

                                       labeledLatLngs state  \
0   [{'label': 'display', 'lat': 30.2671262965541,...    TX   
1   [{'label': 'display', 'lat': 30.26666907980697...    TX   
2   [{'label': 'display', 'lat': 30.26774749787490...    TX   
3   [{'label': 'display', 'lat': 30.26567161471170...    TX   
4   [{'label': 'display', 'lat': 30.26496553144536...    TX   
5   [{'label': 'display', 'lat': 30.26682931530129...    TX   
6   [{'label': 'display', 'lat': 30.26945691311456...    TX   
7   [{'label': 'display', 'lat': 30.26703478951284...    TX   
8   [{'label': 'display', 'lat': 30.26741007051066...    TX   
9   [{'label': 'display', 'lat': 30.24092454292947...    TX   
10  [{'label': 'display', 'lat': 30.262556, 'lng':...    TX   
11  [{'label': 'display', 'lat': 30.24531353894527...    TX   
12  [{'label': 'display', 'lat': 30.269611, 'lng':...    TX   
13  [{'label': 'display', 'lat': 30.26851607432779...    TX   

                          id    Neighborhood  Zipcode   Latitude  Longitude  
0   49b6d215f964a52013531fe3       East Side    78702  30.264515 -97.715185  
1   40b13b00f964a520e8f31ee3       Tarrytown    78703  30.291375 -97.766235  
2   4ba452f8f964a520fe9538e3  South Congress    78704  30.241225 -97.769465  
3   4101a680f964a520640b1fe3         Mueller    78723  30.304778 -97.6

### Nashville and Austin One-Hot Values

In [108]:
nashville_onehot = pd.get_dummies(NashvilleMerged[['Neighborhood']], prefix="", prefix_sep="")
nashville_onehot['Zip Code'] = NashvilleMerged['Zip Code'] 

fixed_columns = [nashville_onehot.columns[-1]] + list(nashville_onehot.columns[:-1])
nashville_onehot = nashville_onehot[fixed_columns]

nashville_grouped = nashville_onehot.groupby('Zip Code').mean().reset_index()
nashville_grouped


Zip Code  12 South  Antioch/Cane Ridge  Belle Meade  Bellevue  Bordeaux  \
0      37013         0                   1            0         0         0   
1      37027         0                   0            0         0         0   
2      37072         0                   0            0         0         0   
3      37076         0                   0            0         0         0   
4      37115         0                   0            0         0         0   
5      37138         0                   0            0         0         0   
6      37201         0                   0            0         0         0   
7      37203         0                   0            0         0         0   
8      37204         1                   0            0         0         0   
9      37205         0                   0            1         0         0   
10     37206         0                   0            0         0         0   
11     37207         0                   0            0         0         0   
12     37208         0                   0            0         0         0   
13     37209         0                   0            0         0         0   
14     37210         0                   0            0         0         0   
15     37211         0                   0            0         0         0   
16     37212         0                   0            0         0         0   
17     37214         0                   0            0         0         0   
18     37215         0                   0            0         0         0   
19     37216         0                   0            0         0         0   
20     37217         0                   0            0         0         0   
21     37218         0                   0            0         0         1   
22     37219         0                   0            0         0         0   
23     37220         0                   0            0         0         0   
24     37221         0                   0            0         1         0   

    Brentwood  Briley Parkway  Donelson  Downtown  ...  Madison  Music Row  \
0           0               0         0         0  ...        0          0   
1           1               0         0         0  ...        0          0   
2           0               0         0         0  ...        0          0   
3           0               0         0         0  ...        0          0   
4           0               0         0         0  ...        1          0   
5           0               0         0         0  ...        0          0   
6           0               0         0         0  ...        0          0   
7           0               0         0         0  ...        0          0   
8           0               0         0         0  ...        0          0   
9           0               0         0         0  ...        0          0   
10          0               0         0         0  ...        0          0   
11          0               0         0         0  ...        0          0   
12          0               0         0         0  ...        0          0   
13          0               0         0         0  ...        0          0   
14          0               0         0         0  ...        0          0   
15          0               0         0         0  ...        0          0   
16          0               0         0         0  ...        0          1   
17          0               0         1         0  ...        0          0   
18          0               0         0         0  ...        0          0   
19          0               0         0         0  ...        0          0   
20          0               1         0         0  ...        0          0   
21          0               0         0         0  ...        0          0   
22          0               0         0         1  ...        0          0   
23          0               0         0         0  ...        0          0   

In [110]:
kclusters = 3
nashville_grouped_clustering = nashville_grouped.drop('Zip Code', 1)
kmeans = KMeans(n_clusters=kclusters, random_state=0).fit(nashville_grouped_clustering)
kmeans.labels_[0:10]

array([0, 0, 0, 0, 0, 0, 0, 0, 2, 0], dtype=int32)

In [115]:
NashvilleMerged

Cluster Labels                                         name  \
0                0     That's CooL Nashville A Live Music Venue   
1                0             Live On The Green Music Festival   
2                0                            Music City Center   
3                0        Country Music Hall of Fame and Museum   
4                0                            Renaissance Hotel   
5                0                 Music City Walk of Fame Park   
6                0  The Grand Ballroom At The Music City Center   
7                0                    Sony/ATV music publishing   
8                2         Music City Star - Riverfront Station   
9                0                          Music City Showcase   
10               0              Renaissance Music City Ballroom   
11               0                           Music City Central   
12               0             country music hall of fame store   
13               0                            City of Nashville   
14               0                 Republic Live Administration   
15               1                   The Pink Pistol Music City   
16               0                             Ryman Auditorium   
17               0                              2nd Avenue Live   
18               0                               Nashville Live   
19               0                            Music City Events   
20               0           Music City Trolley HOP on/off Tour   
21               0                   New Faces of Country Music   
22               0            Just Love Coffee Cafe - Music Row   
23               0             Music City Center Parking Garage   
24               0                          Music City Networks   

                categories                            address  cc       city  \
0                Speakeasy                2309B Franklin Pike  US  Nashville   
1              Music Venue                        1 Public Sq  US  Nashville   
2        Convention Center                      201 5th Ave S  US  Nashville   
3                   Museum                      222 5th Ave S  US  Nashville   
4                    Hotel                    611 Commerce St  US  Nashville   
5                     Park                      121 4th Ave S  US  Nashville   
6        Convention Center                                NaN  US  Nashville   
7                   Office                      424 Church St  US  Nashville   
8            Train Station                      108 1st Ave S  US  Nashville   
9            Souvenir Shop                       323 Broadway  US  Nashville   
10                   Hotel                                NaN  US  Nashville   
11             Bus Station                  400 Charlotte Ave  US  Nashville   
12           Souvenir Shop                                NaN  US  Nashville   
13                    City                                NaN  US        NaN   
14                  Office                                NaN  US  Nashville   
15                Boutique                      218 3rd Ave N  US  Nashville   
16             Music Venue                      116 5th Ave N  US  Nashville   
17                     Bar                      154 2nd Ave N  US  Nashville   
18               Rock Club                      131 2nd Ave N  US  Nashville   
19                Building  808 Broadway, Nashville, TN 37203  US  Nashville   
20  Transportation Service                         5th Avenue  US  Nashville   
21            Concert Hall                    Commerce street  US  Nashville   
22          Breakfast Spot                 1528 Demonbreun St  US  Nashville   
23                 Parking                                NaN  US  Nashville   
24                  Office                     209 10th Ave S  US  Nashville   

          country  distance  \
0   United States      3930   
1   United States       516   
2   United States       665   
3   United States       670   
4   United States       1

### Map of Nashville's KNN Results

In [119]:
Nashville_map_clusters = folium.Map(location= (36.1627,-86.7816), zoom_start=11)
x = np.arange(kclusters)
ys = [i + x + (i * x) ** 2 for i in range(kclusters)]
colors_array = cm.rainbow(np.linspace(0, 1, len(ys)))
rainbow = [colors.rgb2hex(i) for i in colors_array]
markers_colors = []
for lat, lon, pc, cluster in zip(NashvilleMerged['Latitude'], NashvilleMerged['Longitude'], NashvilleMerged['Zip Code'], NashvilleMerged['Cluster Labels']):
    label = folium.Popup(str(pc) + ', ' + ' Cluster ' + str(cluster), parse_html=True)
    folium.CircleMarker([lat, lon],
                        radius=5,
                        popup=label,
                        color=rainbow[cluster-1],
                        fill=True,
                        fill_color=rainbow[cluster-1],
                        fill_opacity=0.7).add_to(Nashville_map_clusters)
       
Nashville_map_clusters

## Nashville's Neighboorhods are relatively uniform, when it comes to having music venues. The only place that stood out against the rest was Zip Code 37211, which is South Nashville.

### Austin One-Hot for Clustering

In [120]:
austin_onehot = pd.get_dummies(AustinMerged[['Neighborhood']], prefix="", prefix_sep="")
austin_onehot['Zip Code'] = AustinMerged['Zipcode'] 

fixed_columns = [austin_onehot.columns[-1]] + list(austin_onehot.columns[:-1])
austin_onehot = austin_onehot[fixed_columns]

austin_grouped = austin_onehot.groupby('Zip Code').mean().reset_index()
austin_grouped

Zip Code  Allandale  Anderson Mill  Bee Cave  East Side  Great Hills  \
0      78702          0              0         0          1            0   
1      78703          0              0         0          0            0   
2      78704          0              0         0          0            0   
3      78723          0              0         0          0            0   
4      78731          0              0         0          0            0   
5      78732          0              0         0          0            0   
6      78738          0              0         1          0            0   
7      78745          0              0         0          0            0   
8      78746          0              0         0          0            0   
9      78748          0              0         0          0            0   
10     78749          0              0         0          0            0   
11     78750          0              1         0          0            0   
12     78757          1              0         0          0            0   
13     78759          0              0         0          0            1   

    Hyde Park  Lake Travis  Mueller  Oak Park  South Congress  Southland Oaks  \
0           0            0        0         0               0               0   
1           0            0        0         0               0               0   
2           0            0        0         0               1               0   
3           0            0        1         0               0               0   
4           1            0        0         0               0               0   
5           0            1        0         0               0               0   
6           0            0        0         0               0               0   
7           0            0        0         0               0               0   
8           0            0        0         0               0               0   
9           0            0        0         0               0               1   
10          0            0        0         1               0               0   
11          0            0        0         0               0               0   
12          0            0        0         0               0               0   
13          0            0        0         0               0               0   

    Sunset Valley  Tarrytown  Westlake  
0               0          0         0  
1               0          1         0  
2               0          0         0  
3               0          0         0  
4               0          0         0  
5               0          0         0  
6               0          0         0  
7               1          0         0  
8               0          0         1  
9               0          0         0  
10              0          0         0  
11              0          0         0  
12              0          0         0  
13              0          0         0

In [121]:
kclusters = 3
austin_grouped_clustering = austin_grouped.drop('Zip Code', 1)
kmeans = KMeans(n_clusters=kclusters, random_state=0).fit(austin_grouped_clustering)
kmeans.labels_[0:10]


array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0], dtype=int32)

In [123]:
AustinMerged.insert(0, 'Cluster Labels', kmeans.labels_)

AustinMerged

Cluster Labels                             name     categories  \
0                0                  Shiner's Saloon            Bar   
1                0                        Speakeasy    Music Venue   
2                0          Driskill Victorian Room    Music Venue   
3                0                The Elephant Room      Jazz Club   
4                0  Cooper's Old Time Pit Bar-B-Que      BBQ Joint   
5                0           Cedar Street Courtyard         Lounge   
6                0                Paramount Theatre  Movie Theater   
7                0               Barracuda Backyard    Music Venue   
8                0              The Chuggin' Monkey            Bar   
9                0            Come And Take It Live    Music Venue   
10               2        Santa Cruz Studio Theater    Music Venue   
11               0                         The ABGB    Beer Garden   
12               0                     The Townsend         Lounge   
13               1                  Stubb's Bar-B-Q      BBQ Joint   

                     address  cc    city        country  distance  \
0     422 Congress Ave Ste D  US  Austin  United States        22   
1           412 Congress Ave  US  Austin  United States        66   
2              604 Brazos St  US  Austin  United States       151   
3           315 Congress Ave  US  Austin  United States       176   
4           217 Congress Ave  US  Austin  United States       256   
5               208 W 4th St  US  Austin  United States       216   
6           713 Congress Ave  US  Austin  United States       269   
7               611 E 7th St  US  Austin  United States       646   
8               219 E 6th St  US  Austin  United States       231   
9        2015 E Riverside Dr  US  Austin  United States      3225   
10             1805 E 7th St  US  Austin  United States      2007   
11          1305 W Oltorf St  US  Austin  United States      3487   
12  718 Congress Ave Ste 100  US  Austin  United States       275   
13          801 Red River St  US  Austin  United States       674   

                                     formattedAddress  \
0   [422 Congress Ave Ste D (at 5th St), Austin, T...   
1   [412 Congress Ave (btwn 4th & 5th St.), Austin...   
2    [604 Brazos St, Austin, TX 78701, United States]   
3   [315 Congress Ave (at 3rd St.), Austin, TX 787...   
4   [217 Congress Ave (at 2nd St), Austin, TX 7870...   
5   [208 W 4th St (Colorado & Lavaca), Austin, TX ...   
6   [713 Congress Ave (7th St.), Austin, TX 78701,...   
7           [611 E 7th St, Austin, TX, United States]   
8   [219 E 6th St (btw Brazos St. & San Jacinto Bl...   
9   [2015 E Riverside Dr, Austin, TX 78741, United...   
10         [1805 E 7th St, Austin, TX, United States]   
11  [1305 W Oltorf St ('Oltorf and the tracks'), A...   
12  [718 Congress Ave Ste 100 (8th St), Austin, TX...   
13  [801 Red River St (at E 8th St), Austin, TX 78...   

                                       labeledLatLngs state  \
0   [{'label': 'display', 'lat': 30.2671262965541,...    TX   
1   [{'label': 'display', 'lat': 30.26666907980697...    TX   
2   [{'label': 'display', 'lat': 30.26774749787490...    TX   
3   [{'label': 'display', 'lat': 30.26567161471170...    TX   
4   [{'label': 'display', 'lat': 30.26496553144536...    TX   
5   [{'label': 'display', 'lat': 30.26682931530129...    TX   
6   [{'label': 'display', 'lat': 30.26945691311456...    TX   
7   [{'label': 'display', 'lat': 30.26703478951284...    TX   
8   [{'label': 'display', 'lat': 30.26741007051066...    TX   
9   [{'label': 'display', 'lat': 30.24092454292947...    TX   
10  [{'label': 'display', 'lat': 30.262556, 'lng':...    TX   
11  [{'label': 'display', 'lat': 30.24531353894527...    TX   
12  [{'label': 'display', 'lat': 30.269611, 'lng':...    TX   
13  [{'label': 'display', 'lat': 30.26851607432779...    TX   

                          id    Neighborhood  Zipcode   Latitude  Longitude  
0   49b6d215f964a52013531fe3       Ea

### Map of Austin's Cluster Values

In [126]:
Austin_map_clusters = folium.Map(location= (30.2672,-97.7431), zoom_start=11)
x = np.arange(kclusters)
ys = [i + x + (i * x) ** 2 for i in range(kclusters)]
colors_array = cm.rainbow(np.linspace(0, 1, len(ys)))
rainbow = [colors.rgb2hex(i) for i in colors_array]
markers_colors = []
for lat, lon, pc, cluster in zip(AustinMerged['Latitude'], AustinMerged['Longitude'], AustinMerged['Zipcode'], AustinMerged['Cluster Labels']):
    label = folium.Popup(str(pc) + ', ' + ' Cluster ' + str(cluster), parse_html=True)
    folium.CircleMarker([lat, lon],
                        radius=5,
                        popup=label,
                        color=rainbow[cluster-1],
                        fill=True,
                        fill_color=rainbow[cluster-1],
                        fill_opacity=0.7).add_to(Austin_map_clusters)
       
Austin_map_clusters

## Austin has a similar distribution of its music venues as Nashville's. There seems to only be one neighborhood, that has any outlying similarities: Great Hills

### Conclusion

Overall the diversity of music venues in each neighborhood for Austin and Nashville are the same. They both had one neighborhood that ended up having more places one can go to see live music than the others. 

The total number of music venues Nashville is 24, while Austin's is 13. The Venue Category of "Music Venue," may not capture everywhere live music is played, but for the purpose of this exploration it will suffice to say that Nashville has more pure "Music Venues." 

More analysis needs to be done against the overall experience at the music venues, types of music played there, and other overlooked values this experiment did not capture.